ui-manchette-with-spacetimechart/src/helpers.ts

```diff
@@ -0,0 +1,168 @@
+import type {
+  OperationalPoint,
+  SpaceTimeChartProps,
+} from '@osrd-project/ui-spacetimechart/dist/lib/types';
+
+import { BASE_OP_HEIGHT, MAX_TIME_WINDOW, MAX_ZOOM_X, MIN_ZOOM_X } from './consts';
+import type {
+  OperationalPointType,
+  ProjectPathTrainResult,
+  StyledOperationalPointType,
+} from '../types';
+import { getHeightWithoutLastWaypoint, msToS } from './utils';
+
+export const calcOperationalPointsToDisplay = (
+  operationalPoints: OperationalPointType[],
+  height: number,
+  isProportional: boolean,
+  zoom: number
+): StyledOperationalPointType[] => {
+  // For proportional display, we only display points that do not overlap with
+  // the last displayed point:
+  if (isProportional && operationalPoints.length > 0) {
+    // We start with the first operational point:
+    const result: StyledOperationalPointType[] = [
+      { ...operationalPoints[0], display: true, styles: {} },
+    ];
+    const from = operationalPoints.at(0)!.position;
+    const to = operationalPoints.at(-1)!.position;
+    const totalDistance = to - from;
+    const heightWithoutFinalOp = getHeightWithoutLastWaypoint(height);
+    let lastDisplayedOP = result[0];
+
```

Comment:
What do you think about initializing `lastVisibleItem` to keep track of the last visible operational point. This will help avoid the need for an additional `findLast` operation, improving the performance of the code.
```suggestion
let lastVisibleItem = result[0];
```

Reply:
I don't get your comment either, if you look line 31 this is already set. The logic that follow is to ensure that the last op from a path is always displayed in case the penultimate had taken priority, during the "available height check".

Can you give me more explanation?